src/main/kotlin/uk/gov/justice/digital/hmpps/nomisprisonerapi/activities/AttendanceService.kt

```diff
@@ -1,12 +1,84 @@
 package uk.gov.justice.digital.hmpps.nomisprisonerapi.activities
 
+import jakarta.transaction.Transactional
+import org.springframework.data.repository.findByIdOrNull
 import org.springframework.stereotype.Service
 import uk.gov.justice.digital.hmpps.nomisprisonerapi.activities.api.CreateAttendanceRequest
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.activities.api.CreateAttendanceResponse
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.data.BadDataException
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.data.ConflictException
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.data.NotFoundException
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.AttendanceOutcome
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.EventStatus
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.OffenderCourseAttendance
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.CourseScheduleRepository
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.OffenderBookingRepository
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.OffenderCourseAttendanceRepository
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.OffenderProgramProfileRepository
+import uk.gov.justice.digital.hmpps.nomisprisonerapi.jpa.repository.ReferenceCodeRepository
 
 @Service
-class AttendanceService {
+@Transactional
+class AttendanceService(
+  private val attendanceRepository: OffenderCourseAttendanceRepository,
+  private val scheduleRepository: CourseScheduleRepository,
+  private val offenderBookingRepository: OffenderBookingRepository,
+  private val offenderProgramProfileRepository: OffenderProgramProfileRepository,
+  private val eventStatusRepository: ReferenceCodeRepository<EventStatus>,
+  private val attendanceOutcomeRepository: ReferenceCodeRepository<AttendanceOutcome>,
+) {
 
-  fun createAttendance(scheduleId: Long, bookingId: Long, createAttendanceRequest: CreateAttendanceRequest) {
-    // TODO SDIT-689 Implement this service
+  fun createAttendance(
+    scheduleId: Long,
+    bookingId: Long,
+    createAttendanceRequest: CreateAttendanceRequest,
+  ): CreateAttendanceResponse =
+    attendanceRepository.save(createAttendanceRequest.toOffenderCourseAttendance(scheduleId, bookingId))
+      .let { CreateAttendanceResponse(it.eventId) }
+
+  private fun CreateAttendanceRequest.toOffenderCourseAttendance(scheduleId: Long, bookingId: Long): OffenderCourseAttendance {
+    val courseSchedule = scheduleRepository.findByIdOrNull(scheduleId)
+      ?: run { throw NotFoundException("Course schedule with id=$scheduleId not found") }
+
+    val offenderBooking = offenderBookingRepository.findByIdOrNull(bookingId)
+      ?: run { throw NotFoundException("Offender booking with id=$bookingId not found") }
```

Comment:
out of interest - do you need run to wrap the exception throwing?

Reply:
No I don't. Only on line 52.

I'll get rid of these in the next PR.